include/nektar_interface/particle_cell_mapping/particle_cell_mapping_newton.hpp

```diff
@@ -0,0 +1,793 @@
+   * required to compute the residual should be contained in the memory region
+   * (i.e. a struct) pointed to by d_data. Computes and returns F(xi) for the
+   * passed xi. Note F should always be of the form X - P, not P - X, such that
+   * the implementation of the residual can be reused as an implementation of X
+   * by passing P=0.
+   *
+   * @param[in] d_data Pointer to data required to perform the Newton iteration.
+   * @param[in] xi0 Current iteration of xi coordinate, x component.
+   * @param[in] xi1 Current iteration of xi coordinate, y component.
+   * @param[in] xi2 Current iteration of xi coordinate, z component.
+   * @param[in] phys0 Target coordinate in physical space, x component.
+   * @param[in] phys1 Target coordinate in physical space, y component.
+   * @param[in] phys2 Target coordinate in physical space, z component.
+   * @param[in, out] f0 F(xi), x component.
+   * @param[in, out] f1 F(xi), y component.
+   * @param[in, out] f2 F(xi), z component.
+   * @returns Residual.
+   */
+  inline REAL newton_residual(const void *d_data, const REAL xi0,
+                              const REAL xi1, const REAL xi2, const REAL phys0,
+                              const REAL phys1, const REAL phys2, REAL *f0,
+                              REAL *f1, REAL *f2) {
+    auto &underlying = static_cast<SPECIALISATION &>(*this);
+    return underlying.newton_residual_v(d_data, xi0, xi1, xi2, phys0, phys1,
+                                        phys2, f0, f1, f2);
+  }
+
+  /**
+   * Get the number of coordinate dimentions the iteration is performed in.
+   * i.e. how many position components should be read from the particle and
+   * reference coordinates written.
+   *
+   *  @returns Number of coordinate dimensions.
+   */
+  inline int get_ndim() {
+    auto &underlying = static_cast<SPECIALISATION &>(*this);
+    return underlying.get_ndim_v();
+  }
+
+  /**
+   * Set the initial iteration prior to the newton iteration.
+   *
+   * @param[in] d_data Pointer to data required to perform the Newton iteration.
+   * @param[in] phys0 Target coordinate in physical space, x component.
+   * @param[in] phys1 Target coordinate in physical space, y component.
+   * @param[in] phys2 Target coordinate in physical space, z component.
+   * @param[in,out] xi0 Input x coordinate to reset.
+   * @param[in,out] xi1 Input y coordinate to reset.
+   * @param[in,out] xi2 Input z coordinate to reset.
+   */
+  inline void set_initial_iteration(const void *d_data, const REAL phys0,
+                                    const REAL phys1, const REAL phys2,
+                                    REAL *xi0, REAL *xi1, REAL *xi2) {
+    auto &underlying = static_cast<SPECIALISATION &>(*this);
+    underlying.set_initial_iteration_v(d_data, phys0, phys1, phys2, xi0, xi1,
+                                       xi2);
+  }
+
+  /**
+   *  Map local coordinate (xi) to local collaposed coordinate (eta).
```

Comment:
```suggestion
   *  Map local coordinate (xi) to local collapsed coordinate (eta).
```
Need a find-replace for this one - see lines 202-204 below and possibly elsewhere

Reply:
Fixed